tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -160,26 +176,30 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 
 	ginkgo.It("Upgrading TiDB Cluster", func() {
 		cluster := newTidbClusterConfig(e2econfig.TestConfig, ns, "cluster", "admin", "")
-		cluster.Resources["pd.replicas"] = "1"
+		cluster.Resources["pd.replicas"] = "3"
 		// TLS only works with PD >= v3.0.5
 		if semver.Compare(cfg.GetTiDBVersionOrDie(), "v3.0.5") >= 0 {
 			cluster.Resources["enableTLSCluster"] = "true"
 		}
-		// deploy
+
+		ginkgo.By("Creating webhook certs and self signing it")
+		svcName := "webhook"
+		certCtx, err := apimachinery.SetupServerCert(ns, svcName)
+		framework.ExpectNoError(err, fmt.Sprintf("unable to setup certs for webservice %s", tests.WebhookServiceName))
+
+		ginkgo.By("Starting webhook pod")
+		webhookPod, svc := startWebhook(f, cfg.E2EImage, ns, svcName, certCtx.Cert, certCtx.Key)
+
+		ginkgo.By("Register webhook")
+		oa.RegisterWebHookAndServiceOrDie(ocfg.WebhookConfigName, ns, svc.Name, certCtx)
+
+		ginkgo.By(fmt.Sprintf("Deploying tidb cluster %s", cluster.ClusterVersion))
 		oa.DeployTidbClusterOrDie(&cluster)
 		oa.CheckTidbClusterStatusOrDie(&cluster)
 		oa.CheckDisasterToleranceOrDie(&cluster)
 
-		cluster.ScalePD(3)
```

Comment:
I removed this because scaling logic has already been tested in another test. Each test focus on its own purpose, then every test will be small and easier to test/debug.